### ISSUE TITLE
Internal: Create a METplus "Release Guide" describing how to build releases for the METplus components.

### ISSUE THREAD

**Issue by JohnHalleyGotway** [dtcenter/METplus#673]
## Describe the Task ##
Please see [METplus meeting notes](https://docs.google.com/document/d/1nkAPFijlJq3X3vuOBVBPgVSzTn6IUMEAO_gnEaLskyg/edit#) from 10/27/2020 for more details. This task is to create a 3rd document for METplus in addition to the user and contributor's guides. This document will describe the process for creating a METplus release and consist of up to 8 chapters:
1. Introduction
2. METplus Release
3. MET Release
4. METdatadb Release
5. METcalcpy Release
6. METplotpy Release (or combine METcalcpy/METplotpy into a single chapter if that makes sense?)
7. METviewer Release
8. METexpress Release

The docs/Release_Guide directory will contain .rst files for each of the items listed above. It should also contain a subdirectory named "release_steps". Each top-level .rst file should be very sparse and consist basically of 3 bulleted lists:
- official release (e.g. vX.Y)
- bugfix release (e.g. vX.Y.Z)
- development release (e.g. vX.Y-betaN or vX.Y-rcN)

Each element in that list should source on .rst files that live in the "release_steps" sub-directory. It'd be cool if that content could be hidden by default, but click on it to expand it. So while we have separate lists for each component, they serve up common content from the "release_steps" directory.

Migrate content from:
Existing wiki page:
   https://github.com/dtcenter/METplus/wiki/Creating-a-new-METplus-release
Creating met-10.0_beta1 release:
   https://github.com/dtcenter/MET/issues/1523#issuecomment-714781834

### Time Estimate ###
4-8 hours for many different people.

### Sub-Issues ###
Consider breaking the task down into sub-issues.
- [ ] *Add a checkbox for each sub-issue here.*
May need to break this down, but not quite sure how to do so yet.

### Relevant Deadlines ###
*List relevant project deadlines here or state NONE.*

### Funding Source ###
2702691

### Assignee ###
- [x] Select **engineer(s)** or **no engineer** required: George (METplus), John HG (MET), Minna (METplotpy/METcalcpy), Tatiana (METviewer), Julie (docs infrastructure), Venita (METdatadb) and Molly (METexpress)
- [x] Select **scientist(s)** or **no scientist** required: Keith (review and advise)

### Labels ###
- [x] Select **component(s)**
- [x] Select **priority**
- [x] Select **requestor(s)**

### Projects and Milestone ###
- [x] Review **projects** and select relevant **Repository** and **Organization** ones or add "alert:NEED PROJECT ASSIGNMENT" label
- [x] Select **milestone** to next major version milestone or "Future Versions"

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [x] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdatadb](https://github.com/dtcenter/METdatadb/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)
Details for each component will live in the common METplus Release Guide. So all changes will be in the METplus repo.

## Task Checklist ##
See the [METplus Workflow](https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [x] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)**, **Project(s)**, **Milestone**, and **Linked issues**
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

**Comment by jprestop** [comment]
I have created branch "feature_673_release_guide"  and have set up a very basic infrastructure for this task.

You can view what exists at this time by visiting:
https://ral.ucar.edu/staff/jpresto/work/METplus_jpresto/Release_Guide/index.html

Please let me know if you have any questions.

**Comment by georgemccabe** [comment]
An initial version of a Release Guide for each project is now available in the METplus repository. These documents will evolve as the process is refined. Closing issue.

**Comment by jprestop** [comment]
Reopening issue as METdatadb does not have a complete Release Guide in place. @venitahagerty 
I will move this to the METplus-4.1.0-beta1 release.

**Comment by JohnHalleyGotway** [comment]
Some additional details that arose in creating the MET 10.0.0 Official Releases which are not currently documented:
- Created a main_v10.0-ref branch for automated testing (Update MET Official Release steps)
- That main_v10.0-ref branch triggered a public RTD build which I manually hid (Update MET Official Release steps)
- Created a milestone for the first bugfix release for the new version: met-10.0.1 (bugfix)  (Update MET Official Release steps)
- Cleaned up stale development branches, deleting old ones from GitHub for previously resolved issues  (Update MET Official Release steps)
- Created a "METplus Version 4.0 Support" project at the DTCenter level  (Update METplus Official Release steps)
- Remove beta# and rc# development releases from the website
- Change existing "recommended" releases to "other" (already included in Official Release Steps)

**Comment by georgemccabe** [comment]
Some notes of additions/changes to the METplus Official Release Guide:

**Create -ref branch for main_vX.Y**

* Create main_v4.0-ref branch, replace version file with new version, replace Quick Search URLs with new version

```
git checkout -b main_v4.0-ref
version=4.0.0
echo $version > metplus/VERSION
sed -i 's|develop/|v$version/|g' docs/Users_Guide/quicksearch.rst
git push origin main_v4.0-ref
```

* Verify main_v4.0-ref Testing workflow creates input data volumes, runs tests and generates truth data volumes
* Once confirmed, create main_v4.0 branch from ref branch:

```
git checkout -b main_v4.0
git push origin main_v4.0
```

**Release notes**
* URL for ReadTheDocs v4.0.0 does not exist until release is created
* Cut release, wait for ReadTheDocs to build v4.0.0 documentation, then
link will work
* Obtain PDF from v4.0.0 ReadTheDocs build and attach it to release.
https://metplus.readthedocs.io/_/downloads/en/v4.0.0/pdf/
* Download and rename to METplus_Users_Guide_v4.0.0.pdf

**Update DTC Website**
* Better instructions to navigate to Downloads page after login
  (Community Code > METplus > Downloads)
* Move instructions to change previously Recommended versions to Other after "Click on Save" step because progress on current release is lost
* Add Link: Link text should be the name of the release and the URL should be the release page that was just created under the GitHub Releases tab. Add i.e. "METplus 4.0.0"

**Update the Documentation on the Web**
* Select the new default branch in the dropdown menu for “Default branch” (e.g. main_v4.0.0) the CLICK SAVE!

**Post Release**
* Create new directory on dtcenter web server for next release data, i.e. v4.1

**Comment by georgemccabe** [comment]
An initial version for a release guide is now available for all METplus components. There are a few notes in the above comments to modify for clarity. Once those changes have been made, this issue could be closed. Please add a comment to this issue if there are any outstanding additions needed to any of the release guides that should be completed before closing this issue.

**Comment by georgemccabe** [comment]
The updates to the METplus Release Guide listed in https://github.com/dtcenter/METplus/issues/673#issuecomment-838646029 have been added. After any updates to the MET Release Guide have been completed, we can close this issue.

**Comment by JohnHalleyGotway** [comment]
I have updated the Release Guide instructions based on the MET version 10.1.0 release. I assume we can and should revise these instructions as needed for future releases. But minor doc-only updates do not require a GitHub issue to be created for each.

I'm going to close this issue now, since we have definitely created the release guide and are now actively using it! Thanks to everyone's input and effort in putting this together. I think its fantastic.

If any of the assignees want to keep this issue open, please feel free to reopen it.
